docs/DeveloperGuide.adoc

```diff
@@ -311,14 +329,146 @@ image::UndoRedoActivityDiagram.png[width="650"]
+the size of the array determines the mode of the command.
+
+The strings are compared to two sets of strings containing the supported categories and orders of the function.
+The string of the category and a boolean representing whether the records are to be reversed will then be passed to
+`ModelManager` to sort the records.
+
+Since the displayed list in the UI is a `FilteredList` which is a wrapper for the underlying list `UniqueRecordList` structure,
+sorting the internal list of records in `versionedFinancialPlanner` will post an event that notifies the UI to update
+the displayed list.
+
+The following sequence diagram shows how the sort operation works:
+
+image::SortSequenceDiagram.png[width:800]
+
+=== List history feature
+==== Current Implementation
+The list mechanism is facilitated by `ModelManager`.
+It represents an in-memory model of the FinancialPlanner and is the component which manages the interactions between the commands and the `VersionedFinancialPlanner`.
+ListCommand calls `ModelManager#updateFilteredRecords` and passes in different predicates depending on the argument mode.
+
+This feature has only one keyword `list` but implements 3 different argument modes to allow users to access multiple versions of the same command.
+The three argument modes are as listed below:
+
+* No Argument mode -- Requires no arguments and returns the entire list of records in the FinancialPlanner.
+* Single Argument mode -- Requires a single date and returns all records containing that date
+* Dual Argument mode -- Requires 2 dates, a `start date` and an `end date`. It returns all records containing dates within the time frame of start date and end date, inclusive of both start date and end date
+The mechanism that facilitates these modes can be found in the `ListCommandParser#parse`. Below is a overview of the mechanism:
+
+. The input given by the user is passed to `ArgumentTokeniser#tokenise` to split the input separated by prefixes.
+. This returns a `ArgumentMultiMap` which contains a map with prefixes as keys and their associated input arguments as the value.
+. The string associated with `d/` is then passed into `ListCommandParser#splitByWhitespace` for further processing and returns an array.
+. The argument mode is determined by the size of this array and the elements are further processed into `Date` objects, before creating and returning a `ListCommand` object.
+
+The `ListCommand` has two constructors which makes use of overloading to reduce code complexity.
+
+* One constructor has no arguments and assigns default predicate for the `FilteredList` in `ModelManager`,
+`PREDICATE_SHOW_ALL_RECORDS` which will show all items in the list.
+* The second constructor takes in 2 `Date` arguments and assigns the predicate `DateIsWithinDateIntervalPredicate` which will only show items within the date interval.
+
+The following sequence diagram shows how the list operation works:
+
+image::ListSequenceDiagram.png[width="800"]
```

Comment:
Very good explanation and good job on illustrating the new feature with the sequence diagram